x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/checkpoint/DataFrameTransformsCheckpointService.java

```diff
@@ -222,6 +224,16 @@ private long getTimeStampForTimeBasedSynchronization(SyncConfig syncConfig, long
         for (ShardStats shard : shards) {
             String indexName = shard.getShardRouting().getIndexName();
             if (userIndices.contains(indexName)) {
+                SeqNoStats seqNoStats = shard.getSeqNoStats();
+                // SeqNoStats could be `null`. This indicates that an `AlreadyClosed` exception was thrown somewhere down the stack
+                // Indicates that the index COULD be closed, or at least that the shard is not fully recovered yet.
+                if (seqNoStats == null) {
```

Comment:
@hendrikmuhs let me know what you think.

It is possible for this to throw an NPE, so to protect against that, I am still throwing a checkpoint exception with the index that failed, and that we were unable to gather `seq_no_stats`. The user making the call should not see any privileged information.

I opted to log the `ShardStats` as a warning. If that is too much information, we will have to change this to figure out how to log actionable information given the `seq_no_stats` being null.